tests/decorators/tests.py

```diff
@@ -274,6 +274,31 @@ def method(self, arg):
 
         self.assertEqual(Test().method(1), 1)
 
+    def test_method_decoration(self):
+        """
+        The method_decorator can be used to decorate a class and its methods.
+        """
+        def deco(func):
+            def _wrapper(*args, **kwargs):
+                return True
+            return _wrapper
+
+        @method_decorator(deco, name="method")
+        class Test(object):
+            def method(self):
+                return False
+
+        self.assertTrue(Test().method())
+
+    def test_invalid_non_callable_attribute_decoration(self):
+        """
+        Using the method_decorator on a non-callable attribute raises an error.
+        """
+        with self.assertRaises(TypeError):
```

Comment:
please use `assertRaisesMessage` to verify this is the `TypeError` we expect (also helps for tracking which tests map to which code).